keps/1136-provisioning-request-support/README.md

```diff
@@ -193,6 +193,33 @@ type ProvisioningRequestConfigSpec struct {
 	// +optional
 	// +kubebuilder:default={backoffLimitCount:3,backoffBaseSeconds:60,backoffMaxSeconds:1800}
 	RetryStrategy *ProvisioningRequestRetryStrategy `json:"retryStrategy,omitempty"`
+
+	// podSetUpdates specifies the update of the workload's PodSetUpdates which
+	// are used to target the provisioned nodes.
+	//
+	// +optional
+	PodSetUpdates *ProvisioningRequestPodSetUpdates `json:"podSetUpdates,omitempty"`
+}
+
+type ProvisioningRequestPodSetUpdates struct {
+	// nodeSelector specifies the list of updates for the NodeSelector.
+	//
+	// +optional
+	NodeSelector []ProvisioningRequestPodSetUpdatesNodeSelector `json:"nodeSelector,omitempty"`
+}
+
+type ProvisioningRequestPodSetUpdatesNodeSelector struct {
+	// key specifies the key for the NodeSelector.
+	//
+	//  +required
+	Key string `json:"key,omitempty"`
+
+	// valueFromProvClassDetail specifies the key of the
+	// ProvisioningRequest.status.provisioningClassDetails from which the value
+	// is used for the update.
+	//
+	// +required
+	ValueFromProvClassDetail string `json:"valueFromProvClassDetail,omitempty"`
```

Comment:
```suggestion
	ValueFromProvisioningClassDetails string `json:"valueFromProvisioningClassDetails,omitempty"`
```

We should use full sentences for APIs.

Reply:
ack, but I think it should be singular `valueFromProvisioningClassDetail`, because this is extracting just one value

Reply:
SGTM

Reply:
To be fair, I think we can clarify in the comment where this points exactly. The options we had so far:
1. `valueFromDetail`
2. `ValueFromProvClassDetail`
3. `valueFromProvisioningClassDetail`
4. `valueFromProvisioningClassDetails`

my personal preference 1>2>3>4,  but I'm ok with any of them, let me know @tenzen-y your preference.

Reply:
I think (3 is the most well-defined name since (3 can represent that this field is built based on provisioningClassDetail.

Reply:
Yes, it is the most descriptive. I guess we need to always keep balance between being descriptive and length, so thus my original proposal `valueFromDetail`. However, it was also proposed by Patryk to have more descriptive version like (2.), so (3.) is not much longer than (2), so I'm ok.